omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs2_0/AlertResource2_0.java

```diff
@@ -0,0 +1,270 @@
+
+		if (rep instanceof DefaultRepresentation) {
+			modelImpl
+					.property(TEXT, new StringProperty())
+					.property(SATISFIED_BY_ANY, new BooleanProperty())
+					.property(ALERT_READ, new BooleanProperty())
+					.property(DATE_TO_EXPIRE, new DateProperty())
+					.property(RECIPIENTS, new ArrayProperty(new RefProperty("#/definitions/AlertRecipientGetRef")));
+		}
+		if (rep instanceof FullRepresentation) {
+			modelImpl
+					.property(TEXT, new StringProperty())
+					.property(SATISFIED_BY_ANY, new BooleanProperty())
+					.property(ALERT_READ, new BooleanProperty())
+					.property(DATE_TO_EXPIRE, new DateProperty())
+					.property(CREATOR, new ObjectProperty())
+					.property(DATE_CREATED, new DateTimeProperty())
+					.property(CHANGED_BY, new ObjectProperty())
+					.property(DATE_CHANGED, new DateTimeProperty())
+					.property(RECIPIENTS, new ArrayProperty(new RefProperty("#/definitions/AlertRecipientGetRef")));
+		}
+		return modelImpl;
+	}
+
+	@Override
+	public Model getCREATEModel(Representation rep) {
+		return new ModelImpl()
+				.property(TEXT, new StringProperty())
+				.property(RECIPIENTS, new ArrayProperty(new RefProperty("#/definitions/AlertRecipientCreate")))
+				.property(SATISFIED_BY_ANY, new BooleanProperty())
+				.property(DATE_TO_EXPIRE, new DateProperty());
+	}
+
+	@Override
+	public Model getUPDATEModel(Representation rep) {
+		return new ModelImpl()
+				.property(TEXT, new StringProperty())
+				.property(RECIPIENTS, new ArrayProperty(new RefProperty("#/definitions/AlertRecipientCreate")))
+				.property(SATISFIED_BY_ANY, new BooleanProperty())
+				.property(DATE_TO_EXPIRE, new DateProperty());
+	}
+
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() throws ResourceDoesNotSupportOperationException {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addRequiredProperty(TEXT);
+		description.addRequiredProperty(RECIPIENTS);
+		description.addProperty(SATISFIED_BY_ANY);
+		description.addProperty(DATE_TO_EXPIRE);
+		return description;
+	}
+
+	@Override
+	public DelegatingResourceDescription getUpdatableProperties() throws ResourceDoesNotSupportOperationException {
+		return getCreatableProperties();
+	}
+
+	@Override
+	public Alert newDelegate() {
+		Alert alert = new Alert();
+		alert.setRecipients(new HashSet<>()); // to prevent NullPointerException
+		return alert;
+	}
+
+	@Override
+	public Alert getByUniqueId(String uniqueId) {
+		boolean includeRetired = true;
+		List<Alert> alerts = Context.getAlertService().getAllAlerts(includeRetired);
+		for (Alert alert : alerts) {
+			if (alert.getUuid().equals(uniqueId)) {
+				return alert;
+			}
+		}
+		return null;
+	}
+
+	@Override
+	public Alert save(Alert alert) {
+		return Context.getAlertService().saveAlert(alert);
+	}
+
+	@Override
+	public void purge(Alert alert, RequestContext context) throws ResponseException {
+		if (alert == null) {
```

Comment:
How would one call purge with a null alert?

Reply:
I'm not sure but I've seen other Resources to include such a null check, for example: `ConceptResource1_8, CohortResource1_8, DrugResource1_8` etc.